theme/story.tsx

```diff
@@ -0,0 +1,50 @@
+import type { Culture, Story } from '@prezly/sdk';
+import type { ReactNode } from 'react';
+
+import { api } from '@/theme-kit';
+
+type Match = { slug: string } | { uuid: string };
+
+export async function match(params: Match) {
+    const { contentDelivery } = api();
+
+    const story = await contentDelivery.story(params);
+
+    return story && { story };
+}
+
+export function resolveLocale({ story }: Props) {
+    return story.culture.code;
+}
```

Comment:
Additionally, a page module may export `resolveLocale()` function to infer the current website locale from a non-`locale` route parameter. Primarily for stories `/[slug]` pages.